ssl/statem/statem_srvr.c

```diff
@@ -2654,6 +2656,7 @@ CON_FUNC_RETURN tls_construct_server_hello(SSL_CONNECTION *s, WPACKET *pkt)
 # endif
             EVP_MD_CTX_free(ctx);
             if (ossl_ech_reset_hs_buffer(s, NULL, 0) != 1) {
+                EVP_MD_CTX_free(ctx);
```

Comment:
You've already a free call above. This would be a doublefree.

Reply:
push below should do that